beacon-chain/rpc/validator_server_test.go

```diff
@@ -40,6 +42,76 @@ func TestValidatorIndex_OK(t *testing.T) {
 	}
 }
 
+func TestValidatorEpochAssignments_CorrectAssignmentsAtEpochBoundary(t *testing.T) {
+	db := internal.SetupDB(t)
+	defer internal.TeardownDB(t, db)
+
+	genesis := b.NewGenesisBlock([]byte{})
+	if err := db.SaveBlock(genesis); err != nil {
+		t.Fatalf("Could not save genesis block: %v", err)
+	}
+	genesisRoot, err := hashutil.HashBeaconBlock(genesis)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var pubKey [96]byte
+	copy(pubKey[:], []byte("0"))
+	if err := db.SaveValidatorIndex(pubKey[:], 0); err != nil {
+		t.Fatalf("Could not save validator index: %v", err)
+	}
+
+	beaconState, err := genesisState(1000)
+	if err != nil {
+		t.Fatalf("Could not setup genesis state: %v", err)
+	}
+
+	if err := db.UpdateChainHead(genesis, beaconState); err != nil {
+		t.Fatalf("Could not save genesis state: %v", err)
+	}
+
+	validatorServer := &ValidatorServer{
+		beaconDB: db,
+	}
+	req := &pb.ValidatorEpochAssignmentsRequest{
+		EpochStart: params.BeaconConfig().GenesisSlot,
+		PublicKey:  pubKey[:],
+	}
+	assignmentsForEpoch0, err := validatorServer.ValidatorEpochAssignments(context.Background(), req)
+	if err != nil {
+		t.Fatalf("Could not fetch assignments for epoch 1: %v", err)
+	}
+
+	lastSlotInEpoch0 := params.BeaconConfig().GenesisSlot + (params.BeaconConfig().SlotsPerEpoch) - 1
+	for beaconState.Slot < lastSlotInEpoch0 {
+		beaconState, err = state.ExecuteStateTransition(
+			beaconState,
+			nil,
```

Comment:
Please add type comment here for readability, otherwise the reader has to look up what you are passing as nil